File: Shared/Exceptions.vb
﻿'*********************************************************************************************
' STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Runtime.Serialization

<Serializable()>
Public NotInheritable Class StockFlowMapDuplicateItemException
    Inherits Exception

    Public Sub New()
        MyBase.New("Duplicate Item Exception")
    End Sub

    Public Sub New(ByVal message As String)
        MyBase.New(message)
    End Sub

    Public Sub New(ByVal message As String, ByVal innerException As Exception)
        MyBase.New(message, innerException)
    End Sub

    Private Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
        MyBase.New(info, context)
    End Sub

End Class

File: src/Shared/Exceptions.cs
// stsim-stockflow: SyncroSim Add-On Package (to stsim) for integrating stocks and flows into state-and-transition simulation models in ST-Sim.
// Copyright © 2007-2024 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System;
using System.Runtime.Serialization;

namespace SyncroSim.STSimStockFlow
{
	[Serializable()]
	public sealed class StockFlowMapDuplicateItemException : Exception
	{
		public StockFlowMapDuplicateItemException() : base("Duplicate Item Exception")
		{
		}

		public StockFlowMapDuplicateItemException(string message) : base(message)
		{
		}

		public StockFlowMapDuplicateItemException(string message, Exception innerException) : base(message, innerException)
		{
		}

		private StockFlowMapDuplicateItemException(SerializationInfo info, StreamingContext context) : base(info, context)
		{
		}
	}
}